### ISSUE TITLE
new rule: prefer-locator-based-method

### ISSUE THREAD

**Issue by timdeschryver** [playwright-community/eslint-plugin-playwright#126]
Hi,

I noticed the Playwright API docs discourages using some page methods in favor of the locator based APIs.

For example: https://playwright.dev/docs/api/class-page#page-is-visible

![image](https://github.com/playwright-community/eslint-plugin-playwright/assets/28659384/d88379a0-aa0c-4a48-b676-e602218de0a0)

I wanted to turn this into an eslint rule, which is how I encountered this library 🫡
If I'm not mistaken this rule doesn't exist yet, and if it's OK I can create a PR to add this as a new rule.

This would look something like this:

# Disallow usage of `page.method()` (`prefer-locator-based-method`)

## Rule Details

Example of **incorrect** code for this rule:

```javascript
const visible = await page.isVisible('button');
                           ~~~~~~~~~ Use locator-based locator.isVisible() instead
```

Examples of **correct** code for this rule:

```javascript
const visible = await page.getByRole('button').isVisible();
```

**Comment by mxschmitt** [comment]
We could got that far and discourage it all-together, since you should in theory never have to use that method and use [toBeVisible()](https://playwright.dev/docs/api/class-locatorassertions#locator-assertions-to-be-visible) instead, since it retries -> no flaky tests.

**Comment by mskelton** [comment]
Duplicate of https://github.com/playwright-community/eslint-plugin-playwright/issues/64

I have some work on this started, just never got it across the finish line: https://github.com/playwright-community/eslint-plugin-playwright/tree/web-first-assertions
